File: Python/models/payee/payee.py
#!/usr/bin/env python3
from ..base import BaseModel, db, uuid

# -------------------------------
# Payee Model
# -------------------------------

class Payee(BaseModel):
    """
    Represents a payee entity.
    """
    __tablename__ = 'payees'

    id = db.Column(db.String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    name = db.Column(db.Text, nullable=False, unique=True)
    transfer_account_id = db.Column(db.String(36), db.ForeignKey('accounts.id'), nullable=True)
    deleted = db.Column(db.Boolean, default=False)
    budget_id = db.Column(db.String(36), db.ForeignKey('budgets.id'), nullable=False)

    transactions = db.relationship('Transaction', back_populates='payee')
    transfer_account = db.relationship('Account')

File: frontend/web_app/lib/models/payee.dart
class Payee {
  final String id;
  final String name;
  final String budgetId;
  final String? transferAccountId;
  final bool deleted;

  Payee({
    required this.id,
    required this.name,
    required this.budgetId,
    required this.deleted,
    this.transferAccountId,
  });

  factory Payee.fromJson(Map<String, dynamic> json) {
    return Payee(
      id: json['id'],
      name: json['name'],
      budgetId: json['budget_id'],
      deleted: json['deleted'] ?? false,
      transferAccountId: json['transfer_account_id'],
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'budget_id': budgetId,
        'deleted': deleted,
        'transfer_account_id': transferAccountId,
      };
}
